app/controllers/users/email_confirmations_controller.rb

```diff
@@ -13,7 +13,14 @@ def create
     private
 
     def email_address
-      @email_address ||= EmailAddress.find_with_confirmation_token(params[:confirmation_token])
+      return @email_address if defined?(@email_address)
+
+      email_address = EmailAddress.find_with_confirmation_token(params[:confirmation_token])
+      if email_address.user.confirmed?
+        @email_address = email_address
+      else
+        @email_address = nil
```

Comment:
Other methods in this class call methods on the result of this method without nil-safe operators. Other methods in this class call methods on the result of this method without nil-safe operators.

Do we need to update those?

https://github.com/18F/identity-idp/blob/4a7fb7c09c126e817ccb9451c37deaad70abe9c1/app/controllers/users/email_confirmations_controller.rb#L44

Reply:
My understanding of the existing implementation is that `email_address` being nil is expected (as a failure condition).  I think it returns nil when the token is not associated with an email address, and this change is mostly extending the cases where we'll return nil.

https://github.com/18F/identity-idp/blob/091f2a2ad6d3b5941b2972c6244182ec724e5307/app/services/email_confirmation_token_validator.rb#L17